File: Java/Silver/1002.java
import java.io.*;
import java.util.*;

public class Main {

    private static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

    public static void main(String[] args) throws IOException {

        int testCaseCount = Integer.parseInt(br.readLine());
        while (testCaseCount > 0) {
            System.out.println(calculatePossiblePositionCount());
            testCaseCount--;
        }


    }

    private static int calculatePossiblePositionCount() throws IOException {
        // x1, y1, r1, x2, y2, r2
        int[] positionsAndDistances = Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();

        double distanceBetween = Math.sqrt(Math.pow(positionsAndDistances[0] - positionsAndDistances[3], 2) + Math.pow(positionsAndDistances[1] - positionsAndDistances[4], 2));

        // 동심원의 경우
        if(distanceBetween == 0 && positionsAndDistances[2] == positionsAndDistances[5]) return -1;
        if(distanceBetween == 0) return 0;

        boolean isTriangle = (distanceBetween < positionsAndDistances[2] + positionsAndDistances[5])
                && (positionsAndDistances[2] < distanceBetween + positionsAndDistances[5])
                && (positionsAndDistances[5] < distanceBetween + positionsAndDistances[2]);

        if (isTriangle) return 2;

        // 일직선인 경우
        boolean isLine = (distanceBetween == positionsAndDistances[2] + positionsAndDistances[5])
                || (positionsAndDistances[2] == distanceBetween + positionsAndDistances[5])
                || (positionsAndDistances[5] == distanceBetween + positionsAndDistances[2]);

        if (isLine) return 1;

        // 접점이 없거나 포함관계의 원인 경우
        return 0;
    }
}

File: Archived2/Kotlin/Silver/1002.kt
import java.io.BufferedReader
import java.io.InputStreamReader
import kotlin.math.pow
import kotlin.math.sqrt


fun main() {

    val br = BufferedReader(InputStreamReader(System.`in`))

    var testCaseCount = br.readLine().toInt()
    while (testCaseCount-- > 0) {
        println(calculatePossiblePositionCount(br))
    }

}

fun calculatePossiblePositionCount(br: BufferedReader): Int {

    // x1, y1, r1, x2, y2, r2
    val positionsAndDistances = br.readLine().split(" ").map { it.toDouble() }

    val distanceBetween = sqrt(
        positionsAndDistances[0].minus(positionsAndDistances[3]).pow(2)
                + positionsAndDistances[1].minus(positionsAndDistances[4]).pow(2)
    )

    val isTriangle = (distanceBetween < positionsAndDistances[2] + positionsAndDistances[5])
            && (positionsAndDistances[2] < distanceBetween + positionsAndDistances[5])
            && (positionsAndDistances[5] < distanceBetween + positionsAndDistances[2])

    val isLine = (distanceBetween == positionsAndDistances[2] + positionsAndDistances[5])
            || (positionsAndDistances[2] == distanceBetween + positionsAndDistances[5])
            || (positionsAndDistances[5] == distanceBetween + positionsAndDistances[2])

    return when {
        distanceBetween == 0.0 && positionsAndDistances[2] == positionsAndDistances[5] -> -1
        distanceBetween == 0.0 -> 0
        isTriangle -> 2
        isLine -> 1
        else -> 0
    }

}